internal/ethapi/transaction_args.go

```diff
@@ -135,6 +135,12 @@ func (args *TransactionArgs) setDefaults(ctx context.Context, b Backend) error {
 			return errors.New(`contract creation without any data provided`)
 		}
 	}
+	// Assign a very high gas limit for cases where an accurate gas limit is not critical,
+	// but need to ensure that gas is sufficient.
+	if infiniteGas {
+		tmp := hexutil.Uint64(math.MaxUint64 / 2)
+		args.Gas = &tmp
```

Comment:
This will overwrite any user-provided gas limit. Please put it under the `if args.Gas == nil` check. Also IMO infiniteGas is not a great name. Maybe `skipGasEstimation`? The other point is that we should set ideally to the RPC gas cap (see ToMessage).

Reply:
Make sense, the comments are addressed in [ee59f15](https://github.com/ethereum/go-ethereum/pull/28846/commits/ee59f15a1069b6eff58eff54754fd3a59ea67620).
And the unit test is passed now.